.github/workflows/ci.yml

```diff
@@ -0,0 +1,46 @@
+name: "CI"
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+# For auto-approving/merging PRs.
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  check:
+    name: "Checks and Tests"
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [14.x, 16.x, 18.x]
+
+    steps:
+      - name: "Run Yarn Basic Checks"
+        uses: SkynetLabs/.github/.github/actions/yarn-basic-checks@master
+        with:
+          node-version: ${{ matrix.node-version }}
+        env:
+          SKYNET_JS_INTEGRATION_TEST_SERVER: "https://skynetpro.net"
```

Comment:
does it make sense to run these on both siasky, free and pro?

Reply:
We want to ensure that skynet-js works well, it shouldn't matter which portal it runs against. I would also avoid tying anything to siasky.net at this point.

Reply:
Why do you think we should run CI on more than one portal? Won't that just increase the CI length and chance of failing, without adding coverage? I picked the most stable portal we have.

Reply:
The point is that these should be stable.

If our portals cannot reliably pass these, that is a problem and we should be working on.

Reply:
To be clear, this CI flow runs on PRs. I think we should test the portals in the daily run, which is what it's for. I have an open ticket to have the daily CI flow run on the free and pro servers.

That would help identify specific servers that are having trouble, while not making it harder for individual PRs to go through.

Reply:
I don't really mind I already approved the PR it was just a suggestion.

I just don't see the big deal with running it on all three because you just catch issues before merging, rather than after merging. Furthermore, you catch issues on the PR itself, and not on the "last couple of PRs". Both of those benefits are definitely nice to have imo.

I don't know how long the CI takes, but I assume it takes a couple of minutes right, running it on all portals doesn't increase the running time up until a point it's not workable.

Lastly: 
> while not making it harder for individual PRs to go through
this is kind of concerning, I think i's mostly due to our issues with portal stability lately? 
But a test suite should not make it hard for a PR to go through.
Even though we kind of have this in `skyd` 😅  but this should be different because there's _way_ less concurrency.

Are there a lot of NDFs in the test suite?